### PR TITLE
Qa 2

### PR DESCRIPTION
**Pull request by academey**

### CODE REVIEW

File: app/components/articleSearch/components/searchItem/infoList.tsx [hunk 1]
```diff
@@ -14,27 +14,21 @@ export interface IInfoListProps {
...
-    document.body.appendChild(textField);
-    textField.select();
-    document.execCommand("copy");
-    textField.remove();
+  const textField = document.createElement("textarea");
+  textField.innerText = DOI;
+  document.body.appendChild(textField);
+  textField.select();
+  document.execCommand("copy");
+  textField.remove();
```
Comment (TylorShin) [inline app/components/articleSearch/components/searchItem/infoList.tsx]: :+1:

File: app/components/auth/signUp/index.tsx [hunk 2]
```diff
@@ -13,6 +13,8 @@ import Icon from "../../../icons";
 import { OAUTH_VENDOR } from "../../../api/auth";
 import { parse } from "qs";
 
+const reactCookie = require("react-cookie");
```
Comment (TylorShin) [inline app/components/auth/signUp/index.tsx]: I suggest to use [store.js](https://github.com/marcuswestin/store.js/).
it's more secure and available for almost every circumstance.

File: app/components/common/searchQueryContent/index.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ const SearchQueryContent = ({
 
   while (parsingIndex !== -1) {
     // if searchQuery does not more exist, it will stop.
-    parsingIndex = upperCaseContent.slice(currentIndex, content.length - 1).indexOf(upperCaseSearchQuery);
+    parsingIndex = upperCaseContent.slice(currentIndex, content.length).indexOf(upperCaseSearchQuery);
```
Comment (TylorShin) [inline app/components/common/searchQueryContent/index.tsx]: lol

File: app/components/auth/signUp/index.tsx (hunk 2 above)
Comment (academey) [reply app/components/auth/signUp/index.tsx]: Yes sir

**Comment by TylorShin** [comment]
:sake:

**Pull request merged**